src/Impl/AmqpMessage.php

```diff
@@ -50,15 +50,16 @@ class AmqpMessage implements InteropAmqpMessage
      * @param string $body
      * @param array  $properties
      * @param array  $headers
+     * @param bool   $redelivered
+     * @param int    $flags
      */
-    public function __construct($body = '', array $properties = [], array $headers = [])
+    public function __construct($body = '', array $properties = [], array $headers = [], $redelivered = false, $flags = self::FLAG_NOPARAM)
```

Comment:
the context is an implementation of abstract factory pattern. You are not supposed to create message,queue,topic or any other objects directly, they must be always created by the context.